### DISCUSSION TITLE
How to draw a line on a canvas element

### DISCUSSION THREAD

**Discussion by zamioku** [webdriverio/webdriverio#9346] [Q&A]
Hi, I am trying to automate signing on a digital signature pad, and I can't seem to find any solution or example using WebdriverIO.

What I've tried so far is using .dragAndDrop() and .performActions() methods but nothing happens in both cases. See the snippet below:

```
let location = await $('#signature').getLocation();
browser.performActions([
            {
                type: 'pointer',
                id: 'cursor1',
                parameters: { pointerType: 'mouse' },
                actions: [
                    { type: 'pointerMove', duration: 0, x: Math.floor(location.x), y: Math.floor(location.y) },
                    { type: 'pointerDown', button: 0 },
                    { type: 'pointerMove', duration: 0, x: Math.floor(location.x + 60), y: Math.floor(location.x + 60) },
                    { type: 'pointerUp', button: 0 },
                ],
            },
        ]);
```

A working solution in Selenium with Java looks like this:

```
Actions act = new Actions(driver);

        act.clickAndHold(canvas).moveByOffset(0,-80).release().build().perform();
        act.clickAndHold(canvas).moveByOffset(20,-80).release().build().perform();
```

Can anyone provide me a simple solution, or has no one encountered this type of issue before? 

**EDIT:** I'm currently using WebdriverIO V7

**Comment by christian-bromann** [comment]
@zamioku can you compare the WebdriverIO and Selenium logs? Also maybe you want to try the new v8 [actions API](https://webdriver.io/docs/api/browser/action) that makes creating these actions easier.

**Reply by zamioku** [reply]
Hi @christian-bromann , I've tried updating to WDIO V8, but now i'm getting a different error and I can't seem to resolve it:
```
[chrome 108.0.5359.98 mac os x #0-1] javascript error: javascript error: Cannot read properties of null (reading 'querySelector')
[chrome 108.0.5359.98 mac os x #0-1]   (Session info: chrome=108.0.5359.98)
[chrome 108.0.5359.98 mac os x #0-1]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```
The browser starts but any element command instantly throws this error, even .waitForExist()

I have not changed any of the selectors, nor did I change  the chrome version.

**Reply by christian-bromann** [reply]
Can you provide the full logs please?

**Reply by zamioku** [reply]
I can't share the code, since it's an internal company application, but after some debugging, the error shown is this:
```

javascript error: javascript error: Cannot read properties of null (reading 'querySelector')
  (Session info: chrome=108.0.5359.98)
    at getErrorFromResponseBody (file:///Users/Sorin/Developer/POC/wdio2/node_modules/webdriver/build/utils.js:194:12)
    at NodeJSRequest._request (file:///Users/Sorin/Developer/POC/wdio2/node_modules/webdriver/build/request/index.js:168:23)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Browser.wrapCommandFn (file:///Users/Sorin/Developer/POC/wdio2/node_modules/@wdio/utils/build/shim.js:72:29)
    at Browser.wrapCommandFn (file:///Users/Sorin/Developer/POC/wdio2/node_modules/@wdio/utils/build/shim.js:72:29)
    at Browser.findElement (file:///Users/Sorin/Developer/POC/wdio2/node_modules/webdriverio/build/utils/index.js:186:20)
    at Browser.$ (file:///Users/Sorin/Developer/wdio2/node_modules/webdriverio/build/commands/browser/$.js:89:17)
    at Browser.wrapCommandFn (file:///Users/Sorin/Developer/POC/wdio2/node_modules/@wdio/utils/build/shim.js:72:29)
[0-1] 2022-12-09T09:18:58.754Z ERROR @wdio/utils:shim: TypeError: Cannot read properties of undefined (reading 'framework')

```

It fails on the first command inside of a cucumber step definition:
`$('>>>.login__sign-in-button').click();`

**Reply by erwinheitzman** [reply]
I have seen someone else mentioning an error with the querySelector as well in Gitter

**Reply by christian-bromann** [reply]
@zamioku mind raising an issue with a reproducible example? Thanks!

**Reply by zamioku** [reply]
I've raised a bug, feel free to look into it when you have time:
https://github.com/webdriverio/webdriverio/issues/9367
